pandas/tests/scalar/timestamp/test_constructors.py

```diff
@@ -34,6 +34,14 @@ def test_construct_from_string_invalid_raises(self):
         with pytest.raises(ValueError, match="gives an invalid tzoffset"):
             Timestamp("200622-12-31")
 
+    def test_constructor_str_infer_reso(self):
+        # non-iso8601 path
+        ts = Timestamp("2016-01-01 1:30:01 PM")
+        assert ts.unit == "s"
+
+        ts = Timestamp("2016 June 3 15:25:01.345")
+        assert ts.unit == "ms"
```

Comment:
I think we need 3 tests here, one for each path:
- `dt = _parse_delimited_date(date_string, dayfirst, out_bestunit)`
- `_parse_dateabbr_string(`
- `dt = dateutil_parse(date_string, default=_DEFAULT_DATETIME,`

Reply:
good idea, updated